From 9dd80ee69aa1cd8c0bc0fe276fd8f10c7363b944 Mon Sep 17 00:00:00 2001
From: Michael Adler <michael.adler@siemens.com>
Date: Mon, 27 Mar 2023 09:53:01 +0200
Subject: [PATCH] refactor: remove setuptools from install_requires

Remove setuptools from install_requires as per PEP 518 [1]. Including
setuptools is mainly for documentation purposes, but it cannot satisfy
the requirement if it's not already satisfied (since setup.py imports
setuptools).

In this particular case, it's even worse, since setuptools is listed in
install_requires, but install_requires is used to specify the
dependencies required to run the package, while setup_requires is used
to specify the dependencies required to run the setup script itself.

[1] https://peps.python.org/pep-0518/

Signed-off-by: Michael Adler <michael.adler@siemens.com>
---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index a1bbe25..0978529 100644
--- a/setup.py
+++ b/setup.py
@@ -22,7 +22,6 @@ setup(
     },
     install_requires=[
         "libconf~=2.0.1",
-        "setuptools~=57.0.0",
     ],
     python_requires=">=3.6",
 )
-- 
2.39.2

